src/fheroes2/army/army_ui_helper.cpp

```diff
@@ -44,14 +44,46 @@ void fheroes2::drawMiniMonsters( const Troops & troops, int32_t cx, const int32_
         count = troops.GetOccupiedSlotCount();
     }
 
-    const int chunk = width / count;
+    int chunk = static_cast<int32_t>( width / count );
+    size_t slots = troops.Size();
+    int slotsToSkip = 1;
+
+    int slotOffset = 1;
 
     if ( !isCompact ) {
-        cx += chunk / 2;
+        if ( count <= 2 ) {
+            slotsToSkip = 2;
+        }
+        else if ( count >= 7 ) {
+            slotsToSkip = 0;
+        }
+
+        int marginLeft = 18;
+        chunk = static_cast<int32_t>( ( width - marginLeft ) / ( count + ( slotsToSkip * 2 ) ) );
```

Comment:
Here we can also make variable types consistent and cast only `count` which is `uint32_t`:
```suggestion
        int32_t marginLeft = 18;
        chunk = ( width - marginLeft ) / ( static_cast<int32_t>( count ) + ( slotsToSkip * 2 ) );
```